### ISSUE TITLE
Add Outlier Score (trajectory-based anomaly detection) to Vipassana

### ISSUE THREAD

**Issue by tamatebox** [tamatebox/samadhi_model#14]
## Summary

Add two additional scores to Vipassana for richer anomaly detection capabilities, complementing the existing Trust Score.

## Motivation

Current Trust Score only uses 8 static Grounding Metrics → good for OOD detection but not for detecting **known but minority patterns** (e.g., fraud that converges properly with `label_guidance=True`).

The GRU-encoded `dynamic_context` is already computed but only flows to `V_ctx` for the Decoder. It's not used for scoring.

### Triple Score Architecture

| Score | Input | Purpose | Example Use |
|:---|:---|:---|:---|
| **trust_score** (existing) | metrics (8 dims) | OOD detection | New fraud patterns |
| **conformity_score** (new) | dynamic_context (GRU) | Majority pattern matching | Known fraud patterns |
| **confidence_score** (new) | metrics + dynamic_context | Comprehensive assessment | Final decision |

### Design Rationale: Pure Separation

Each score uses **distinct inputs** to enable clear interpretation:
- `trust_score`: Only static metrics → "Is the result normal?"
- `conformity_score`: Only dynamic context → "Is the process normal?"
- `confidence_score`: Both → "Overall, is this normal?"

Mixing inputs in trust/conformity would cause correlation and blur the 4-quadrant interpretation.

### 4-Quadrant Interpretation

```
              Trust Score (Result)
                High        Low
           ┌──────────┬──────────┐
Conformity │ Normal   │ OOD      │
(Process)  │ (typical)│ (unknown)│
High       ├──────────┼──────────┤
           │ Process  │ Total    │
Low        │ Anomaly  │ Anomaly  │
           └──────────┴──────────┘

+ confidence_score for comprehensive judgment
```

### Implementation

```python
# In StandardVipassana._build_networks()

# Existing: trust_head (metrics → trust_score)
self.trust_head = nn.Sequential(
    nn.Linear(self.NUM_METRICS, 2 * self.NUM_METRICS),
    nn.ReLU(),
    nn.Linear(2 * self.NUM_METRICS, 1),
    nn.Sigmoid(),
)

# New: conformity_head (dynamic_context → conformity_score)
self.conformity_head = nn.Sequential(
    nn.Linear(self.gru_hidden_dim, self.gru_hidden_dim // 2),
    nn.ReLU(),
    nn.Linear(self.gru_hidden_dim // 2, 1),
    nn.Sigmoid(),
)

# New: confidence_head (both → confidence_score)
self.confidence_head = nn.Sequential(
    nn.Linear(self.gru_hidden_dim + self.NUM_METRICS, (self.gru_hidden_dim + self.NUM_METRICS) // 2),
    nn.ReLU(),
    nn.Linear((self.gru_hidden_dim + self.NUM_METRICS) // 2, 1),
    nn.Sigmoid(),
)
```

### Loss Function (Stage 2)

All three scores share the same target:
```python
loss = BCE(trust_score, target) + BCE(conformity_score, target) + BCE(confidence_score, target)
```

Targets:
- Clean: `1.0`
- Augmented: `1.0 - severity`
- Drunk/Mismatch/Void: `0.0`

Same target, different inputs → each head learns different judgment criteria.

## Implementation Plan

1. Add `conformity_head` and `confidence_head` to `StandardVipassana`
2. Update `forward()` to return all three scores
3. Update `BaseVipassana` interface
4. Update Stage 2 training loss
5. Update docs: specification_en.md, theory_en.md, training_strategy.md
6. Add config option to enable/disable new heads (backward compatibility)

## Backward Compatibility

- Existing `trust_score` unchanged
- New scores are additive (opt-in via config)
- Default behavior remains single trust_score

## Computational Cost

Additional ~5K parameters (negligible):
- conformity_head: ~2K params
- confidence_head: ~2.6K params

**Comment by tamatebox** [comment]
## Final Design Decision

### Loss Function (Stage 2)

Equal weights for all three scores:

```python
loss = BCE(trust_score, target) + BCE(conformity_score, target) + BCE(confidence_score, target)
```

### Rationale

1. All three scores learn the same task (normal/anomaly detection) from different perspectives
2. No reason for one to dominate over others
3. Simpler to start with, can adjust later if needed

### Config Option (for future tuning)

```python
# StandardVipassanaConfig
trust_weight: float = 1.0
conformity_weight: float = 1.0  
confidence_weight: float = 1.0
```

### Summary of Triple Score

| Score | Input | Gradient to GRU | Purpose |
|:---|:---|:---|:---|
| trust_score | metrics | No | OOD detection (result-based) |
| conformity_score | dynamic_context | **Yes** | Pattern conformity (process-based) |
| confidence_score | both | **Yes** | Comprehensive assessment |

This design:
1. Fixes the GRU training issue (currently not trained)
2. Provides richer anomaly interpretation
3. Maintains backward compatibility (scores are additive)

**Comment by tamatebox** [comment]
Closing as implemented. Triple Score System (trust_score, conformity_score, confidence_score) is now part of v4.1.
